authority/provisioner/provisioner.go

```diff
@@ -150,7 +151,16 @@ func (l *List) UnmarshalJSON(data []byte) error {
 		case "azure":
 			p = &Azure{}
 		default:
-			return errors.Errorf("provisioner type %s not supported", typ.Type)
+			// Skip unsupported provisioners. A client using this method may be
+			// compiled with a version of smallstep/certificates that does not
+			// support a specific provisioner type. If we don't skip unknown
+			// provisioners, a client encountering an unknown provisioner will
+			// break. Rather than break the client, we skip the provisioner but
+			// warn the user that an unknown provisioner was found and suggest
+			// that the user update their client's dependency on
+			// step/certificates and recompile.
+			log.Printf("[WARN] Unknown provisioner of type '%s' found. Please update your client.", typ.Type)
```

Comment:
In step-sds we do a provisioner request to get a JWK provisioner, and we don't care about the others, so, an error like the following one doesn't make much sense:
```
2019/08/29 12:55:23 [WARN] Unknown provisioner of type 'GCP' found. Please update your client.
```
And it's also a server, but is acting as a client at this point. It makes sense for `step-ca` and might make sense for `step`, so having some kind of pluggable logger might be better.